env0/resource_agent_project_assignment.go

```diff
@@ -57,138 +43,99 @@ func resourceAgentProjectAssignment() *schema.Resource {
 	}
 }
 
-func resourceAgentProjectAssignmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	var newAssignment AgentProjectAssignment
-	if err := readResourceData(&newAssignment, d); err != nil {
-		return diag.Errorf("schema resource data deserialization failed: %v", err)
-	}
-
-	apaLock.Lock()
-	defer apaLock.Unlock()
-
+func resourceAgentProjectAssignmentCreateOrUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	apiClient := meta.(client.ApiClientInterface)
-	assignments, err := apiClient.ProjectsAgentsAssignments()
-	if err != nil {
-		return diag.Errorf("could not get project-agent assignments: %v", err)
-	}
 
-	for projectId, agentId := range assignments.ProjectsAgents {
-		if projectId == newAssignment.ProjectId && agentId == newAssignment.AgentId {
-			return diag.Errorf("assignment for project id %v and agent id %v already exist", projectId, agentId)
-		}
+	var assignment AgentProjectAssignment
+	if err := readResourceData(&assignment, d); err != nil {
+		return diag.Errorf("schema resource data deserialization failed: %v", err)
 	}
 
-	assignments.ProjectsAgents[newAssignment.ProjectId] = newAssignment.AgentId
+	payload := client.AssignProjectsAgentsAssignmentsPayload{
+		assignment.ProjectId: assignment.AgentId,
+	}
 
-	if _, err := apiClient.AssignAgentsToProjects(assignments.ProjectsAgents); err != nil {
-		return diag.Errorf("could not update project-agent assignments: %v", err)
+	if _, err := apiClient.AssignAgentsToProjects(payload); err != nil {
+		return diag.Errorf("failed to assign project '%s' to agent '%s': %v", assignment.ProjectId, assignment.AgentId, err)
 	}
 
-	d.SetId(newAssignment.GetId())
+	d.SetId(assignment.ProjectId)
```

Comment:
Does it matter much if the ID is the project ID or assignment ID?

Reply:
I don't think an assignment has an id.
So AFAIK it has to be the project_id (assuming one assignment per project).

Reply:
Up until now this was `GetId`, which is a combo of agentKey and project ID. And now it has changed. Wouldn't this affect existing configurations, causing drift or other issues?

Reply:
I guess ProjectId makes more sense here as the ID. What would be the implications of this change on existing configurations?

Reply:
yes you are correct.

The feature did not work as expected earlier, so fixing it now may cause a drift.
I think the best thing we can do is just add a note in the release notes about this breaking change. (That's what we've done in the past).

Reply:
Alright. Could you please test the behaviour locally so we'd know exactly what users might see once they upgrade?